DQM/SiPixelPhase1TrackEfficiency/src/SiPixelPhase1TrackEfficiency.cc

```diff
@@ -90,20 +96,34 @@ void SiPixelPhase1TrackEfficiency::analyze(const edm::Event& iEvent, const edm::
       bool isHitMissing = hit->getType()==TrackingRecHit::missing;
 
       const SiPixelRecHit* pixhit = dynamic_cast<const SiPixelRecHit*>(hit->hit());
-      int row = 0, col = 0;
+      const PixelGeomDetUnit* geomdetunit = dynamic_cast<const PixelGeomDetUnit*> ( tracker->idToDet(id) );
+      const PixelTopology& topol = geomdetunit->specificTopology();
+      LocalPoint lp;
       if (pixhit) {
-        const PixelGeomDetUnit* geomdetunit = dynamic_cast<const PixelGeomDetUnit*> ( tracker->idToDet(id) );
-        const PixelTopology& topol = geomdetunit->specificTopology();
-        LocalPoint const& lp = pixhit->localPositionFast();
-        MeasurementPoint mp = topol.measurementPosition(lp);
-        row = (int) mp.x();
-        col = (int) mp.y();
+        lp = pixhit->localPosition();
+      } else {
+        TrajectoryStateCombiner tsc;
+        TrajectoryStateOnSurface tsos = tsc(measurement.forwardPredictedState(), measurement.backwardPredictedState());
```

Comment:
Thanks -- though this does not help w.r.t. #17098? I think I found a suitable value in TrackExtra, but this cannot go into this PR since it is not in CMSSW yet...